File: Tests/CustomerDataTests.fs
module Tests.CustomerDataTests

open System
open Xunit


open System.IO
open System
open System.Collections.Generic
open SomeBasicCassandraApp
open GetCommands
open Cassandra

let commands = getCommands()
let cluster = Cluster.Builder()
                     .AddContactPoint("localhost")
                     .WithDefaultKeyspace("cassandratests")
                     .Build()
// Connect to the nodes using a keyspace
let session = cluster.ConnectAndCreateDefaultKeyspaceIfNotExists()
let repo = CassandraRepository(session) :> IRepository
let commandHandler = Commands.handle repo
for command in commands |> Array.map WithSeqenceNumber.getCommand do
  commandHandler command
[<Fact>]
let CanGetCustomerById()=
  Assert.NotNull(repo.GetCustomer(1))

[<Fact>]
let CanGetProductById()=
  Assert.NotNull(repo.GetProduct(1))

[<Fact>]
let OrderContainsProduct()=
  let order = repo.GetOrder(1)
  Assert.True(order.Products |> Array.tryFind( fun p -> p = 1) |> Option.isSome)

//[<Fact>]
//member this.OrderHasACustomer()=
//    Assert.IsNotNullOrEmpty(_repository.GetTheCustomerOrder(1).Firstname)

File: Tests/CustomerDataTests.cs
using System.Linq;
using Core.Domain;
using Xunit;

namespace Tests
{
    public class CustomerDataTests
    {
        private static IRepository _repository;
        private static ObjectContainer _container;
        static CustomerDataTests()
        {
            _container = new ObjectContainer();
            //_container.Boot();
            _repository = _container.GetRepository();
            var commands = new GetCommands().Get();
            _container.HandleAll(commands);
        }
        [Fact]
        public void CanGetCustomerById()
        {
            Assert.NotNull(_repository.GetCustomer(1));
        }

        [Fact]
        public void CanGetProductById()
        {
            Assert.NotNull(_repository.GetProduct(1));
        }
        [Fact]
        public void OrderContainsProduct()
        {
            Assert.True(_repository.GetOrder(1).Products.Contains(1));
        }
        /*[Fact]
        public void OrderHasACustomer()
        {
            Assert.True(!string.IsNullOrEmpty( _repository.GetTheCustomerOrder(1).Firstname));
        }*/
       
    }
}
